backend/helpers/handleError.js

```diff
@@ -0,0 +1,52 @@
+/*
+ANIME BENSALEM, BRIAHANA MAUGÉ, JOSEPH P. PASAOA
+Server Error Handling Helper | Capstone App (Pursuit Volunteer Mgr)
+*/
+
+
+const errorResponse = require('./errorResponse');
+
+/*
+USAGE
+to instantiate custom error, format: throw new Error(`<error code>__error: <msg>`);
+example: throw new Error(`404__error: volunteer ${volunteerId} not found`);
+*/
+
+
+const handleError = (err, req, res, next) => {
+  // per documentation, checking for present headers and if so, passing
+  if (res.headersSent) {
+    console.log("err: res headers already exist. passing error to express");
+    return next(err);
+  }
+
+  // catch and parse error.messages for relevant http code
+  let [ code, msg = code ] = err.message.split('__');
+
+  // if "__" wasn't found, error was not thrown from a customized instance
+  if (msg === code) {
+    errorResponse(res, 500, `(back) error: ${msg}`);
+  } else {
+
+    // specific-case errors
+    if (err.message === "No data returned from the query.") {
```

Comment:
Here also you will end up by sending more then one response (please refer to my second comment (sorry, I started by that one). 
I think an `else` will be a quick fix

Reply:
switched in reversion to alter err.message and pass through instead of sending response